src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+
+		JButton copyButton = new JButton("Copy");
+		copyButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				ClipboardUtilities.put(textArea.getText());
+			}
+		});
+		copyButton.setToolTipText("Copy text into clipboard");
+		panel.add(copyButton);
+
+		JPanel bottomPane = new JPanel();
+		FlowLayout flowLayout = (FlowLayout) bottomPane.getLayout();
+		flowLayout.setHgap(125);
+		add(bottomPane, BorderLayout.SOUTH);
+
+		JButton submitRunButton = new JButton("Submit");
+		submitRunButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+			    submitSelectedJudgesSolutions();
+			}
+		});
+
+		JButton resetButton = new JButton("Reset");
+		resetButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				resetFields(true);
+			}
+		});
+		resetButton.setToolTipText("Reset back to default settings");
+		bottomPane.add(resetButton);
+		submitRunButton.setToolTipText("Submit selected sample runs");
+		bottomPane.add(submitRunButton);
+
+//		centerPane2.add(controlsPane, BorderLayout.CENTER);
+		JScrollPane leftSide = new JScrollPane(controlsPane);
+		
+		JPanel submitNoSamplesPane = new JPanel();
+		submitNoSamplesPane.setBorder(new TitledBorder(null, "", TitledBorder.LEADING, TitledBorder.TOP, null, null));
+		FlowLayout fl_submitNoSamplesPane = (FlowLayout) submitNoSamplesPane.getLayout();
+		fl_submitNoSamplesPane.setAlignment(FlowLayout.LEFT);
+		submitNoSamplesPane.setBounds(33, 131, 650, 32);
+		controlsPane.add(submitNoSamplesPane);
+
+		checkBoxSubmitFailingSamples = new JCheckBox("Submit Failing Samples");
+		checkBoxSubmitFailingSamples.setVerticalAlignment(SwingConstants.TOP);
+		submitNoSamplesPane.add(checkBoxSubmitFailingSamples);
+		checkBoxSubmitFailingSamples.setSelected(true);
+		checkBoxSubmitFailingSamples.setToolTipText("Submt all non-AC (Yes) submissions");
+
+		judgingTypesButtonGroup.add(submitAllJudgingTypesRadioButton);
+		submitAllJudgingTypesRadioButton.setSelected(true);
+		judgingTypesButtonGroup.add(submitSelectedJudgingTypeRadioButton);
+		
+		submitNoSamplesPane.add(submitAllJudgingTypesRadioButton);
+		submitNoSamplesPane.add(submitSelectedJudgingTypeRadioButton);
+		
+		leftSide.setPreferredSize(new Dimension(230, 230));
+		leftSide.setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_AS_NEEDED);
+		leftSide.setWheelScrollingEnabled(true);
+		splitPane.setLeftComponent(leftSide);
+
+		// centerPane2.add(LogPanel, BorderLayout.SOUTH);
+		JScrollPane logScrollPanel = new JScrollPane(LogPanel);
+		logScrollPanel.setPreferredSize(new Dimension(60, 60));
+		logScrollPanel.setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_AS_NEEDED);
+		logScrollPanel.setWheelScrollingEnabled(true);
+		splitPane.setRightComponent(logScrollPanel);
+		
+		
+		selectedNosLabel.setToolTipText("None selected");
+		selectedNosLabel.addMouseListener(new MouseAdapter() {
```

Comment:
Adding a MouseListener to a JLabel is likely to be confusing -- there's nothing to tell the user that the label has a potential action associated with it.  Since the "action" here is simply to show a list of the Judgement Types, I suggest that instead of adding a MouseListener to the JLabel it would be better to put a little "?" clickme info icon next to the label; that's a common mechanism (especially in multiple different PC2 GUI components) to get additional information  such as what the MouseListener here is trying to convey.